v03_pipeline/lib/reference_data/compare_globals.py

```diff
@@ -15,131 +19,98 @@
-) -> bool:
-    dataset_config = CONFIG[dataset][reference_genome.v02_value]
-    dataset_ht = import_ht_from_config_path(dataset_config, reference_genome)
-    checks = {
-        'version': ht_version_matches_config(
-            joined_ht_globals,
-            dataset,
-            dataset_config,
-            dataset_ht,
-        ),
-        'path': ht_path_matches_config(joined_ht_globals, dataset, dataset_config),
-        'enum': ht_enums_match_config(joined_ht_globals, dataset, dataset_config),
-        'select': ht_selects_match_config(
-            joined_ht,
-            dataset,
-            dataset_config,
-            dataset_ht,
-        ),
-    }
-
-    results = []
-    for check, result in checks.items():
-        if result is False:
-            logger.info(f'{check} mismatch for {dataset}')
-        results.append(result)
-    return all(results)
-
-
-def ht_version_matches_config(
-    joined_ht_globals: ReferenceDataGlobals,
-    dataset: str,
-    dataset_config: dict,
-    dataset_ht: hl.Table,
-) -> bool:
-    joined_ht_version = joined_ht_globals.versions.get(dataset)
-    if joined_ht_version is None:
-        return False
-
-    config_or_dataset_version = hl.eval(
-        parse_dataset_version(
-            dataset_ht,
-            dataset,
-            dataset_config,
-        ),
-    )
-    return joined_ht_version == config_or_dataset_version
-
-
-def ht_path_matches_config(
-    joined_ht_globals: ReferenceDataGlobals,
-    dataset: str,
-    dataset_config: dict,
-) -> bool:
-    joined_ht_path = joined_ht_globals.paths.get(dataset)
-    if joined_ht_path is None:
-        return False
-
-    config_path = get_ht_path(dataset_config)
-    return joined_ht_path == config_path
-
-
-def ht_enums_match_config(
-    joined_ht_globals: ReferenceDataGlobals,
-    dataset: str,
-    dataset_config: dict,
-) -> bool:
-    joined_ht_enums = joined_ht_globals.enums.get(dataset, {})
-    config_enums = dataset_config.get('enum_select', {})
-    return joined_ht_enums == config_enums
-
-
-def ht_selects_match_config(
-    joined_ht: hl.Table,
-    dataset: str,
-    dataset_config: dict,
-    dataset_ht: hl.Table,
-) -> bool:
-    joined_ht_selects = set(joined_ht[dataset])
-    config_selects = set(get_all_select_fields(dataset_ht, dataset_config).keys())
-    return len(config_selects.symmetric_difference(joined_ht_selects)) == 0
+                result = self.ht1_globals[field.name].get(dataset) == self.ht2_globals[
+                    field.name
+                ].get(dataset)
+            if result is False:
+                logger.info(f'{field.name} mismatch for {dataset}, {self.rdc.value}')
+            results.append(result)
+        return all(results)
+
+    def _compare_selects(self, dataset: str) -> bool:
+        ht1_selects = self.ht1_globals.selects.get(dataset)
+        ht2_selects = self.ht2_globals.selects.get(dataset)
+        if ht1_selects is None or ht2_selects is None:
```

Comment:
If both of these are `None` we would want to return `True` here right?  But realistically neither should ever be `None`?

Reply:
I added this here because the `symmetric_difference` operation throws an error if one of the sets is `None`, however you're right that if they're both `None` we'd want to return `True`. 

The case where one is `None` and one is a set would happen when we're creating the rdc table for the first time, where the `Globals` attributes for the rdc table would be empty. I can't think of a case where they'd both be `None` irl

Reply:
defaulting to an empty set handles these cases

Reply:
awesome! cool.  I think that makes sense.  I was wondering if we could just make use of `set`  equality in python here.

In that case we might not need to special case `_compare_selects`?

```
>>> set([3,2,2]) == {2,3}
True
```

Reply:
you're right it really is that simple!!